### PR TITLE
Revise Result Op

### PR DESCRIPTION
**Pull request by blesniewski**
### Details:
 - *Added tests:*
     - *type prop*
     - *visitor (attributes)*
     - *SLT(CPU instantiation)*
     - *SSLT*
     - *backend*

### Tickets:
 - *52867*
 - *53937*
 - *52868*
 - *52864*

### CODE REVIEW

**Review by pelszkow — APPROVED** [review]

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/result.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#include <vector>
+
+#include "shared_test_classes/single_layer/result.hpp"
+
+using namespace LayerTestsDefinitions;
+using namespace InferenceEngine;
+
+namespace {
+TEST_P(ResultLayerTest, Serialize) {
```
Comment (ggalieroc) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/result.cpp:13]: I guess this is a typo from serialization tests. Small reminder for single layer tests instantiation with Bucketize example [here](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/shared/include/single_layer_tests/bucketize.hpp)

**Review by ilyachur — COMMENTED** [review]
Can we replace `Result::type_info` by `NGRAPH_RTTI_DEFINITION`?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/result.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "shared_test_classes/single_layer/result.hpp"
+
+using namespace LayerTestsDefinitions;
+using namespace InferenceEngine;
+
+namespace {
+TEST_P(ResultLayerTest, CompareWithRefs) {
+    Run();
+}
```
Comment (ggalieroc) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/result.cpp:15]: Please move this block to header file in inference-engine/tests/functional/plugin/shared/include/single_layer_tests

File: inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/result.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace LayerTestsDefinitions {
+
+using ConfigMap = typename std::map<std::string, std::string>;
+
+using ResultTestParamSet = typename std::tuple<
+    InferenceEngine::SizeVector,          // Input shapes
+    InferenceEngine::Precision,           // Input precision
+    LayerTestsUtils::TargetDevice,        // Device name
+    ConfigMap>;                           // Additional network configuration
```
Comment (ggalieroc) [inline inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/result.hpp:24]: Is this param used at all?

**Review by ilyachur — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

File: inference-engine/tests/functional/shared_test_classes/src/single_layer/result.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+void ResultLayerTest::SetUp() {
+    std::vector<size_t> inputShape;
+    InferenceEngine::Precision inputPrecision;
+    std::string targetDevice;
+    ConfigMap additionalConfig;
+    std::tie(inputShape, inputPrecision, targetDevice, additionalConfig) = GetParam();
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inputPrecision);
```
Comment (ilyachur) [inline inference-engine/tests/functional/shared_test_classes/src/single_layer/result.cpp:30]: Am I right that you create a function:
```
Const      Parameter
     |
    \/
Result
```
Why do you need to create such network? Why do you need to have redundant operations?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**